### DISCUSSION TITLE
Expected behavior of using -dump-from option on a thread

### DISCUSSION THREAD

**Discussion by pican79** [rusq/slackdump#66] [Q&A] [answered]
Hi,
when I use the `-dump-from` option on a thread, it dumps all messages (and all files if I specify the `-f` option), including the ones that are before the `-dump-from` timestamp.

Is it because the thread "start timestamp" is before the `-dump-from` timestamp?
I'm using the v2.0.0-beta.2 version.
Thanks.

**Comment by rusq** [comment]
@pican79 I'll have to check the code to answer, I'll let you know.  As far as I remember, it should only get the messages from the timestamp that is specified in `--dump-from`, but I might be mistaken.

**Comment by rusq** [comment, marked as answer]
@pican79 the changes has been merged to master, now `-dump-from` will restrict the threads as well.

**Reply by pican79** [reply]
Thanks for the fix. I validated it with the [v2.0.0-rc.2 release](https://github.com/rusq/slackdump/releases/tag/v2.0.0-rc.2).
